src/framework/theme/components/datepicker/datepicker.directive.ts

```diff
@@ -367,6 +373,16 @@ export class NbDatepickerDirective<D> implements OnDestroy, ControlValueAccessor
           this.hidePicker();
         }
       });
+
+    merge(
+      this.picker.blur,
+      fromEvent(this.input, 'blur').pipe(
+        filter(() => !this.picker.isShown && this.document.activeElement !== this.input),
+      ),
+    ).pipe(
+      takeWhile(() => this.alive),
+      take(1),
```

Comment:
Why do we need `take(1)` here? As I remember the stream will be completed after one item was emitted in this case, no?